test_handler.go

```diff
@@ -73,10 +83,33 @@ func testHandler(w http.ResponseWriter, r *http.Request) {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
-    testRunsJSON := string(testRunsBytes)
 
-    if err := templates.ExecuteTemplate(w, "index.html", testRunsJSON); err != nil {
+    data := struct {
+        TestRuns string
+        SHA      string
+    }{
+        string(testRunsBytes),
+        runSHA,
+    }
+
+    if err := templates.ExecuteTemplate(w, "index.html", data); err != nil {
         http.Error(w, err.Error(), http.StatusInternalServerError)
         return
     }
 }
+
+func getRunSHA(r *http.Request) (runSHA string, err error) {
+    // Get the SHA for the run being loaded (the first part of the path.)
+    runSHA = "latest"
+    params, err := url.ParseQuery(r.URL.RawQuery)
+    if err != nil {
+        return
+    }
+
+    runParam := params.Get("sha")
+    regex := regexp.MustCompile("[0-f]{10}")
+    if regex.MatchString(runParam) {
+        runSHA = runParam
+    }
+    return
```

Comment:
I think this should be `return runSHA`

Reply:
that's the same (naked returns in Go... not a fan of them TBH so made it explicit)